src/v2/cookbook/using-axios-to-consume-apis.md

```diff
@@ -0,0 +1,173 @@
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
+
+This is a lot easier for us to display, so we can now update our html to display only the information we need from the data we've received, and we'll create a [filter](../api/#Vue-filter) to make sure that the decimal is in the appropriate place as well.
+
+```html
+<div id="app">
+  <h1>Bitcoin Price Index</h1>
+  <div v-for="currency in info" class="currency">
+    {{ currency.description }}:
+    <span class="lighten">
+      <span v-html="currency.symbol"></span>{{ currency.rate_float | currencydecimal }}
+    </span>
+  </div>
+</div>
+```
+
+```js
+filters: {
+  currencydecimal(value) {
+    return value.toFixed(2)
+  }
+},
+```
+
+<p data-height="300" data-theme-id="32763" data-slug-hash="9d59319c09eaccfaf35d9e9f11990f0f" data-default-tab="result" data-user="Vue" data-embed-version="2" data-pen-title="Third Step Axios and Vue" class="codepen">See the Pen <a href="https://codepen.io/team/Vue/pen/9d59319c09eaccfaf35d9e9f11990f0f/">Third Step Axios and Vue</a> by Vue (<a href="https://codepen.io/Vue">@Vue</a>) on <a href="https://codepen.io">CodePen</a>.</p>
+<script async src="https://static.codepen.io/assets/embed/ei.js"></script>
+
+### Dealing with Errors
+
+There are times when we might not get the data we need from the API. There are several reasons that our axios call might fail, including but not limited to:
+
+* The API is down.
+* The request was made incorrectly.
+* The API isn't giving us the information in the format that we anticipated.
+
+When making this request, we should be checking for just such circumstances, and giving ourselves information in every case so we know how to handle the problem. In an axios call, we'll do so by using `catch`.
+
+```js
+axios
+  .get('https://api.coindesk.com/v1/bpi/currentprice.json')
+  .then(response => (this.info = response.data.bpi))
+  .catch(error => console.log(error))
+```
+
+This will let us know if something failed during the API request, but what if the data is mangled or the API is down? Right now the user will just see nothing. We might want to build a loader for this case, and then tell the user if we're not able to get the data at all.
+
+```js
+new Vue({
+  el: '#app',
+  data() {
+    return {
+      info: null,
+      loading: true,
+      errored: false
+    }
+  },
+  filters: {
+    currencydecimal(value) {
+      return value.toFixed(2)
+    }
+  },
+  mounted() {
+    axios
+      .get('https://api.coindesk.com/v1/bpi/currentprice.json')
+      .then(response => {
+        this.loading = false
+        this.info = response.data.bpi
+      })
+      .catch(error => {
+        console.log(error)
+        this.errored = true
+        this.loading = false
+      })
```

Comment:
Can we extract `this.loading = false` into a `finally` clause instead?

Reply:
yes, good idea